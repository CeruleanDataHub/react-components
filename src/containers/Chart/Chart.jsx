import Highcharts from "highcharts";
import HighchartsReact from "highcharts-react-official";
import HighchartsWheel from "highcharts/modules/dependency-wheel";
import HighchartSankey from "highcharts/modules/sankey";
import TreeMap from "highcharts/modules/treemap";
import PropTypes from "prop-types";
import React from "react";

HighchartSankey(Highcharts);
HighchartsWheel(Highcharts);
TreeMap(Highcharts);

const chartDefaults = {
  credits: { enabled: false }
};

export const Chart = ({
  title,
  xAxis,
  series,
  type = "line",
  options = {}
}) => {
  const chartOptions = {
    ...chartDefaults,
    ...options,
    xAxis,
    series,
    title: { ...options.title, text: title },
    chart: { ...options.chart, type },
    plotOptions: {
      ...options.plotOptions,
      series: {
        animation: false
      }
    }
  };

  return <HighchartsReact highcharts={Highcharts} options={chartOptions} />;
};

Chart.propTypes = {
  /** Title of the chart */
  title: PropTypes.string,
  /** Values for X axis */
  xAxis: PropTypes.arrayOf(PropTypes.shape({ categories: PropTypes.array })),
  /** Data to be graphed */
  series: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      data: PropTypes.array
    })
  ),
  /** Type of the chart */
  type: PropTypes.oneOf([
    "line",
    "bar",
    "area",
    "spline",
    "areaspline",
    "heatmap"
  ]),
  /** You can use highcharts options to extend the chart behavior */
  // eslint-disable-next-line react/forbid-prop-types
  options: PropTypes.object
};

Chart.defaultProps = {
  title: "",
  xAxis: null,
  series: [],
  type: "line",
  options: {}
};
