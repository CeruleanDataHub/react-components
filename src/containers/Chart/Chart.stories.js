import React from "react";

import { Chart } from "./Chart";

export default {
  title: "Chart",
  component: Chart,
};

const xAxis = [
  {
    categories: [
      "monday",
      "tuesday",
      "wednesday",
      "thursday",
      "friday",
      "saturday",
      "sunday",
    ],
  },
];

const series = [{ name: "temperature", data: [10, 15, 20, 25, 15, 20, 15] }];

export const ChartStory = () => <Chart xAxis={xAxis} series={series} />;
