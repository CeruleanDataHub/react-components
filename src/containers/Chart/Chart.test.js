import { shallow } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";

import { Chart } from "./Chart";

describe("Chart", () => {
  it("should render chart", () => {
    const component = renderer.create(<Chart />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render chart with simple data", () => {
    const xAxis = [
      {
        categories: [
          "monday",
          "tuesday",
          "wednesday",
          "thursday",
          "friday",
          "saturday",
          "sunday",
        ],
      },
    ];

    const series = [
      { name: "temperature", data: [10, 15, 20, 25, 15, 20, 15] },
    ];

    const component = shallow(<Chart xAxis={xAxis} series={series} />);

    expect(component).toMatchSnapshot();
  });
});
