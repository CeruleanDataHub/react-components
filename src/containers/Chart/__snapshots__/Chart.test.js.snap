// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart should render chart 1`] = `
<div>
   loading ... 
</div>
`;

exports[`Chart should render chart with simple data 1`] = `ShallowWrapper {}`;
